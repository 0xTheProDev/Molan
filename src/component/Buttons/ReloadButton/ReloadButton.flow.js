// @flow
export type ReloadButtonProps = {
  ReloadButtonText: string,
  onReload: (event: MouseEvent) => void,
};
