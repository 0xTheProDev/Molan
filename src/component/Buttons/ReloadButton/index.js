// @flow
export { ReloadButton as default } from './ReloadButton';
export { ReloadButtonProps } from './ReloadButton.flow';
