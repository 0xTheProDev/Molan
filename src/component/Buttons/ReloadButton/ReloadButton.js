// @flow
import React from 'react';
import { Button, Icon } from 'semantic-ui-react';
import { noop } from 'util/common';

import { ReloadButtonProps } from './ReloadButton.flow';

export function ReloadButton({ ReloadButtonText, onReload }: ReloadButtonProps) {
  return (
    <Button
      animated
      floated="right"
      title={ReloadButtonText}
      onClick={onReload}
    >
      <Button.Content hidden>{ReloadButtonText}</Button.Content>
      <Button.Content visible>
        <Icon name="redo" />
      </Button.Content>
    </Button>
  );
}

ReloadButton.defaultProps = {
  ReloadButtonText: 'Reload',
  onReload: noop,
};
