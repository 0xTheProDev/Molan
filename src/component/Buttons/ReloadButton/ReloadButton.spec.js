import React from 'react';
import { Button } from 'semantic-ui-react';

import { ReloadButton } from './ReloadButton';

describe('Component::Buttons:ReloadButton', () => {
  let
    minProps,
    onReloadStub;

  beforeAll(() => {
    onReloadStub = jest.fn();
    minProps = {
      ReloadButtonText: 'Reload',
      onReload: onReloadStub,
    };
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should render with base structure', () => {
    const wrapper = shallow(<ReloadButton {...minProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke callback prop when button clicked', () => {
    const wrapper = shallow(<ReloadButton {...minProps} />);
    wrapper.find(Button).simulate('click');
    expect(onReloadStub).toHaveBeenCalled();
  });
});
