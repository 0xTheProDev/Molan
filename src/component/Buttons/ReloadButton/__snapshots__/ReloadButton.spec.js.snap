// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::Buttons:ReloadButton should render with base structure 1`] = `
<Button
  animated={true}
  as="button"
  floated="right"
  onClick={[MockFunction]}
  title="Reload"
>
  <ButtonContent
    hidden={true}
  >
    Reload
  </ButtonContent>
  <ButtonContent
    visible={true}
  >
    <Icon
      as="i"
      name="redo"
    />
  </ButtonContent>
</Button>
`;
